server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -1946,6 +1951,65 @@ public void testEnableSearchWorkerThreads() throws IOException {
         }
     }
 
+    public void testEnableQueryPhaseParallelCollection() throws IOException {
+        IndexService indexService = createIndex("index", Settings.EMPTY);
+        IndexShard indexShard = indexService.getShard(0);
+        ShardSearchRequest request = new ShardSearchRequest(
+            OriginalIndices.NONE,
+            new SearchRequest().allowPartialSearchResults(randomBoolean()),
+            indexShard.shardId(),
+            0,
+            indexService.numberOfShards(),
+            AliasFilter.EMPTY,
+            1f,
+            System.currentTimeMillis(),
+            null
+        );
+        int executorPoolSize = randomIntBetween(1, 100);
+        ExecutorService threadPoolExecutor = new ThreadPoolExecutor(
```

Comment:
I remember hitting forbidden API issues around creating the executor this way. Look how the other tests do it.

Reply:
Thanks!